api/pkg/db/model/model.go

```diff
@@ -21,19 +21,19 @@ type (
 
 	Catalog struct {
 		gorm.Model
-		Name       string
-		Type       string
-		URL        string
-		Owner      string
+		Name       string `gorm:"size:100;not null;unique"`
+		Type       string `gorm:"size:100;not null;unique"`
+		URL        string `gorm:"size:100;not null;unique"`
```

Comment:
why is this unique? wouldn't this prevent 2 branches of the same repo to be added a s catalog?

Reply:
IMHO, we should only add constraints that are absolutely required by the domain we are trying to map.